src/transformStateWithClones.js

```diff
@@ -7,7 +7,54 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const FINAL_ARR = [];
+  let modObj = { ...state };
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        modObj = addProps(modObj, obj.extraData);
+        break;
+
+      case 'removeProperties':
+        modObj = removeProps(modObj, obj.keysToRemove);
+        break;
+
+      case 'clear':
+        modObj = clearObj(modObj);
+    }
+    FINAL_ARR.push(modObj);
+  }
+
+  return FINAL_ARR;
+}
+
+function addProps(data, extraData) {
+  const OBJ = { ...data };
+
+  Object.assign(OBJ, extraData);
+
+  return OBJ;
+}
+
+function removeProps(data, keyToRemove) {
+  const OBJ = { ...data };
```

Comment:
![image](https://github.com/mate-academy/js_stateful-clones/assets/86923089/45063544-8c83-4cd1-9347-a3b50a26755f)

It meant **really** constant values, not values defined using `const` keyword - FYI